src/briefcase/integrations/subprocess.py

```diff
@@ -44,70 +44,112 @@ def final_kwargs(self, **kwargs):
     def run(self, args, **kwargs):
         """A wrapper for subprocess.run()
 
-        If verbosity >= 2, the executed command will be printed to the console.
-
         The behavior of this method is identical to subprocess.run(),
         except for the `env` argument. If provided, the current system
         environment will be copied, and the contents of env overwritten
         into that environment.
         """
         # Invoke subprocess.run().
         # Pass through all arguments as-is.
-        # All exceptions are propegated back to the caller.
-        if self.command.verbosity >= 2:
-            print(">>> {cmdline}".format(
-                cmdline=' '.join(shlex.quote(str(arg)) for arg in args)
-            ))
+        # All exceptions are propagated back to the caller.
+        self._log_command(args)
+        self._log_environment(kwargs)
 
-        return self._subprocess.run(
-            [
-                str(arg) for arg in args
-            ],
-            **self.final_kwargs(**kwargs)
-        )
+        try:
+            command_result = self._subprocess.run(
+                [
+                    str(arg) for arg in args
+                ],
+                **self.final_kwargs(**kwargs)
+            )
+        except subprocess.CalledProcessError as exc:
+            self._log_return_code(exc.returncode)
+            raise
+
+        self._log_return_code(command_result.returncode)
+        return command_result
 
     def check_output(self, args, **kwargs):
         """A wrapper for subprocess.check_output()
 
-        If verbosity >= 2, the executed command will be printed to the console.
-
         The behavior of this method is identical to subprocess.check_output(),
         except for the `env` argument. If provided, the current system
         environment will be copied, and the contents of env overwritten
         into that environment.
         """
-        # Invoke subprocess.check_output
-        if self.command.verbosity >= 2:
-            print(">>> {cmdline}".format(
-                cmdline=' '.join(shlex.quote(arg) for arg in args)
-            ))
+        self._log_command(args)
+        self._log_environment(kwargs)
 
-        return self._subprocess.check_output(
-            [
-                str(arg) for arg in args
-            ],
-            **self.final_kwargs(**kwargs)
-        )
+        try:
+            cmd_output = self._subprocess.check_output(
+                [
+                    str(arg) for arg in args
+                ],
+                **self.final_kwargs(**kwargs)
+            )
+        except subprocess.CalledProcessError as exc:
+            self._log_output(exc.output, exc.stderr)
```

Comment:
I agree with potential for confusion from potentially disjointed output in the event of an error.

What may be possible here is to always force `stderr` to be redirected to `stdout` for `check_output` calls. Although, I am not sure if that's even safe or possible as is; it may be best suited for that to be evaluated more in depth within an effort to generalize parsing return values from `check_output`.